.github/workflows/ros2_build.yml

```diff
@@ -0,0 +1,52 @@
+name: Test ROS2 Build Linux
+
+on:
+  push:
+    branches-ignore:
+      - 'doc*'
+      - 'wip*'
+  pull_request:
+    branches:
+      - '*'
+
+jobs:
+  call_core_build:
+    uses: ./.github/workflows/core_build.yml
```

Comment:
If the `ros2_humble_test` was just a job inside `full_build`, we only need to call `call_core_build` once. Now it is called twice.
I would thus move this job to the `full_build` file. Can you do that? Or is there a strong reason to not do that?